h2o-algos/src/test/java/hex/deeplearning/DeepLearningGradientCheck.java

```diff
@@ -63,7 +63,7 @@ public void map(Chunk[] c) {
               DistributionFamily.laplace,
               DistributionFamily.quantile,
               DistributionFamily.huber,
-              DistributionFamily.modified_huber,
+              // DistributionFamily.modified_huber,
```

Comment:
Modified_huber is something that doesn't work properly right now, so we don't want people to actually use it; while at the same time we don't want to remove it completely because one day we hope to get around to fixing it. 
Hence I'm just disabling some of the tests that were trying to use this distribution, plus adding an assertion check in the model parameter validation to prevent such models from being built.